[Feat] Render lidar before agents and allow for tuple as color

Zartris opened the pull request:
![image](https://github.com/proroklab/VectorizedMultiAgentSimulator/assets/14197299/f72ce46c-2249-4098-931e-487abd6d1017)
Her is an image of my problem. 

When lidar is rendered after the agent geom, it is hard to see the shape of the agent. 

I have two methods for solving this.
1. Render lidar before the agent, hence we can see the agent being rendered on top.
2. Like in the Agent class, we make a getter function for color, that checks if it is an Enum or a tuple. This allows the user to specify an alpha on the lidar color. The downside of this is the user needs to understand that this color tuple needs to be 3 or 4 values, whereas before this was more restricted. But as this is the same as we do in the Agent class I do not see the big harm.

These are small changes that should not break anything.

matteobettini commented on vmas/simulator/core.py:
```diff
@@ -1045,11 +1045,14 @@ def render(self, env_index: int = 0) -> "List[Geom]":
         if self._sensors is not None:
             for sensor in self._sensors:
                 geoms += sensor.render(env_index=env_index)
+
+        for geom in geoms:
+            geom.set_color(*self.color, alpha=self._alpha)
+
```
What you are doing is now setting the colors of the sensor geoms to that of the agent

matteobettini commented on vmas/simulator/sensors.py:
```diff
@@ -6,12 +6,13 @@
 import torch
 
 import vmas.simulator.core
 import vmas.simulator.utils
+from vmas.simulator.utils import Color
```
Have you checked this works? 

the reason the imports above are so general is because importing specific elements was causing me circular imports error

matteobettini replied:
```suggestion
from vmas.simulator.utils import Color
```

matteobettini commented on vmas/simulator/sensors.py:
```diff
@@ -125,7 +132,7 @@ def render(self, env_index: int = 0) -> "List[Geom]":
 
                 ray_circ = rendering.make_circle(0.01)
-                ray_circ.set_color(*self._render_color.value)
+                ray_circ.set_color(*self.render_color)
```
This just affects the ray circ and not the line, your problem seems to be the line

matteobettini reviewed (commented):
I am not so keen of changing the order of rendering objects (unless strictly necessary) as this might affect other users that were relying on sensors being rendered on top of the agent.

I am keen to add an alpha to sensors geoms, but we have to decide a way to uniform this.

Currently:
- agents have self.color and self.alpha
- entities have just self.color (which you just made me notice that it can sontrol alpha if it is size 4)

I think we should make also entities and sensors have self._alpha and force colors to be either the vmas object or a 3 dimentional tuple or tensor

Zartris commented:
1. Removed the render order change, so now it is back to being agent first then lidar
2. Changed so now it is indeed the ray and not the circle color. 
3 .Added alpha in construct, changed it from a tuple of 4 to a tuple of 3.

matteobettini commented on vmas/simulator/sensors.py:
```diff
@@ -123,9 +136,9 @@ def render(self, env_index: int = 0) -> "List[Geom]":
                 xform.set_translation(*self.agent.state.pos[env_index])
                 xform.set_rotation(angle)
                 ray.add_attr(xform)
+                ray.set_color(*self.render_color, alpha=self.alpha)
```
```suggestion
                ray.set_color(*Color.BLACK.value, alpha=self.alpha)
```
rays are black

matteobettini commented on vmas/simulator/sensors.py:
```diff
@@ -123,9 +136,9 @@ def render(self, env_index: int = 0) -> "List[Geom]":
                 ray.add_attr(xform)
+                ray.set_color(*self.render_color, alpha=self.alpha)
 
                 ray_circ = rendering.make_circle(0.01)
```
```suggestion
                ray_circ = rendering.make_circle(0.01)
                ray_circ.set_color(*self.render_color, alpha=self.alpha)
```
the color of the lidar is the color of its circles, we set it with the same alpha as the lines

matteobettini commented on vmas/simulator/sensors.py:
```diff
@@ -123,9 +136,10 @@ def render(self, env_index: int = 0) -> "List[Geom]":
                 xform.set_translation(*self.agent.state.pos[env_index])
                 xform.set_rotation(angle)
                 ray.add_attr(xform)
+                ray.set_color(*Color.BLACK.value, alpha=self.alpha)
```
```suggestion
                ray.set_color(r=0, g=0, b=0, alpha=self.alpha)
```
The default black is not 0,0,0, so to keep the color it was before we have to use this

matteobettini reviewed (approved):
Perfect thanks!

These last changes and we are good to go
